packages/commonwealth/client/scripts/views/pages/NotificationSettings/CommentSubscriptionEntry.tsx

```diff
@@ -0,0 +1,134 @@
+import { CommentSubscription } from '@hicommonwealth/schemas';
+import { getThreadUrl, safeTruncateBody } from '@hicommonwealth/shared';
+import { notifySuccess } from 'controllers/app/notifications';
+import { pluralize } from 'helpers';
+import { getRelativeTimestamp } from 'helpers/dates';
+import { useCommonNavigate } from 'navigation/helpers';
+import React, { useCallback } from 'react';
+import { Link } from 'react-router-dom';
+import { useDeleteCommentSubscriptionMutation } from 'state/api/trpc/subscription/useDeleteCommentSubscriptionMutation';
+import { getCommunityUrl } from 'utils';
+import { CWCommunityAvatar } from 'views/components/component_kit/cw_community_avatar';
+import { CWText } from 'views/components/component_kit/cw_text';
+import { CWThreadAction } from 'views/components/component_kit/new_designs/cw_thread_action';
+import { QuillRenderer } from 'views/components/react_quill_editor/quill_renderer';
+import { User } from 'views/components/user/user';
+import { z } from 'zod';
+
+interface CommentSubscriptionEntryProps {
+  readonly subscription: z.infer<typeof CommentSubscription>;
+  readonly onUnsubscribe: (id: number) => void;
+}
+
+export const CommentSubscriptionEntry = (
+  props: CommentSubscriptionEntryProps,
+) => {
+  const { subscription, onUnsubscribe } = props;
+  const comment = subscription.Comment!;
+  const thread = comment.Thread;
+  const comment_id = comment.id!;
+
+  const deleteCommentSubscriptionMutation =
+    useDeleteCommentSubscriptionMutation();
+
+  const threadUrl = getThreadUrl(
+    {
+      chain: thread.community_id,
+      id: comment.id!,
+      title: thread.title,
+    },
+    comment_id,
+    true,
+  );
+
+  const deleteThreadSubscription = useCallback(async () => {
+    await deleteCommentSubscriptionMutation.mutateAsync({
+      id: `${comment_id}`,
+      comment_ids: [comment_id],
+    });
+  }, [deleteCommentSubscriptionMutation, comment_id]);
+
+  const handleDeleteSubscription = useCallback(() => {
+    deleteThreadSubscription()
+      .then(() => {
+        notifySuccess('Unsubscribed!');
+        onUnsubscribe(comment_id);
+      })
+      .catch(console.error);
+  }, [deleteThreadSubscription, onUnsubscribe, comment_id]);
+
+  const navigate = useCommonNavigate();
+
+  const handleNavigateToThread = useCallback(() => {
+    navigate(threadUrl);
+  }, [navigate, threadUrl]);
```

Comment:
Nit: maybe not use callbacks (it will use memory), as this is a simple function call